src/tools/imageSrcset.js

```diff
@@ -0,0 +1,48 @@
+export default class {
+    getSrcset(url, sizes) {
+        // Regex to test size string is of the form 123x123 or 100w
+        const sizeRegex = /(^\d+w$)|(^(\d+?)x(\d+?)$)/;
+        // Regex to test to that srcset descriptor is of the form 1x 1.5x 2x OR 123w
+        const descriptorRegex = /(^\d+w$)|(^([0-9](\.[0-9]+)?)x)$/;
+
+        function isObject(obj) {
```

Comment:
How strict do we need to be that object is object?  Some JavaScript utility libraries have used typeof var == 'object' and object != null as it is considered faster.  Other libraries have used, Object.prototype.toString.call(var) === '[object Object]' as it is considered the more reliable way to typecheck.

Reply:
This runs in the FE so there are fewer security concerns than vs a handlebars helper.

I'm mostly concerned with making sure people aren't passing ints or arrays. Hopefully people will observe Cornerstone's implementation against this method and copy it.